flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptor.java

```diff
@@ -120,7 +125,7 @@ public TaskDeploymentDescriptor(
 			List<InputGateDeploymentDescriptor> inputGates,
 			List<BlobKey> requiredJarFiles, int targetSlotNumber) {
 
-		this(jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks,
+		this(jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks, 0,
```

Comment:
I think such constructors with default values are dangerous, as people tend to call the wrong constructors.
Someone who creates the deployment descriptor should think about providing the attempt number.